sycl/test-e2e/AddressSanitizer/out-of-bounds/buffer/buffer.cpp

```diff
@@ -0,0 +1,34 @@
+// REQUIRES: linux, cpu
+// RUN: %{build} %device_asan_flags -O0 -g -o %t.out
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t.out 2>&1 | FileCheck %s
+// RUN: %{build} %device_asan_flags -O1 -g -o %t.out
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t.out 2>&1 | FileCheck %s
+// RUN: %{build} %device_asan_flags -O2 -g -o %t.out
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t.out 2>&1 | FileCheck %s
+
+#include <sycl/detail/core.hpp>
+
+static const int N = 16;
+
+int main() {
+  sycl::queue q;
+
+  std::vector<int> v(N);
+  for (int i = 0; i < N; i++)
+    v[i] = i;
+
+  {
+    sycl::buffer<int, 1> buf(v.data(), v.size());
```

Comment:
nit: I'd write like this:

```
sycl::buffer<int, 1> buf(N);
{
  sycl::host_accessor acc{buf};
  std::iota(acc.begin(), acc.end(), 0);
}
```
```

Reply:
Thanks for suggestions. I write like this because we want to test the pattern that buffer uses host ptr.

Reply:
Please add an explicit comment describing that. Otherwise it's possible someone would try to "refactor" your tests by using `host_accessor` as above.

Reply:
Done